src/ExtractGQL.ts

```diff
@@ -263,10 +264,22 @@ export class ExtractGQL {
       kind: 'Document',
       definitions: [],
     };
-    retDocument.definitions = document.definitions.filter((definition: DefinitionNode) => {
+    retDocument.definitions = document.definitions.reduce((carry: FragmentDefinitionNode[], definition: DefinitionNode) => {
       const definitionName = (definition as (FragmentDefinitionNode | OperationDefinitionNode)).name;
-      return (isFragmentDefinition(definition) && queryFragmentNames[definitionName.value] === 1);
-    });
+      if ((isFragmentDefinition(definition) && queryFragmentNames[definitionName.value] === 1)) {
+        const definitionExists = carry.findIndex(
+          (value: FragmentDefinitionNode) => value.name.value === definitionName.value
+        ) !== -1;
+
+        // If this definition doesn't exist yet, add it.
+        if (!definitionExists) {
+          return [...carry, definition];
+        }
+      }
+
+      return carry;
+    }, ([] as FragmentDefinitionNode[])).sort(sortFragmentsByName);
```

Comment:
This `sort` doesn't stand out to me. IMO the reduce function should be defined separately so this line could be
```
   retDocument.definitions = document.definitions.reduce(reduceQueryDefinitions, ([] as FragmentDefinitionNode[])).sort(sortFragmentsByName);
```